API.Test/ReviewControllerTest.cs

```diff
@@ -0,0 +1,240 @@
+﻿using Moq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Reviews.API.Controllers;
+using Reviews.API.DTOs;
+using Reviews.API.Services;
+using Reviews.Domain.Entities;
+using Reviews.Domain.Exceptions;
+using Reviews.Domain.Entities.Factories;
+using Reviews.Infrastructure.Kafka;
+using Reviews.Infrastructure.Repositories;
+
+namespace API.Tests
+{
+    public class ReviewControllerTests
+    {
+        [Fact]
+        public async Task CreateReview_RestaurantReview_ShouldReturnOkResult()
+        {
+            // Arrange
+            var mockKafkaProducer = new Mock<IKafkaProducer>();
+            var mockRepo = new Mock<IReviewRepository>();
+
+            var mockFactoryResolver = new Mock<Func<string, IReviewFactory>>();
+            mockFactoryResolver
+                .Setup(resolver => resolver("restaurant"))
+                .Returns(new RestaurantReviewFactory());
+
+            var reviewService = new ReviewService(mockKafkaProducer.Object, mockFactoryResolver.Object, mockRepo.Object);
+            var mockLogger = new Mock<ILogger<ReviewController>>();
+            var controller = new ReviewController(mockLogger.Object, reviewService);
+
```

Comment:
Jeg ville nok lægge alle var'sne op over testsne, så du ikke behøver lave nye hver gang - evt tjek mine accounts tests for info. 
Så det ser lidt mere overskueligt ud. ( Altså alt der ligger i arrange som du gentager i hver test )